### PR TITLE
Branch selection

### PR DESCRIPTION
**Pull request by rdvelazquez**
This is a branch selection component that should be usable for datamonkey, the gui and on galaxy. We can decide where the code should live when we work on integrating datamonkey and the gui. For now all the gui code, shared or otherwise, is going in this repo

Still to do for this Pull Request:
 - [x] Allow selecting reference branches for RELAX
 - [x] Evaluate if it is feasible to test components that use d3, and to test that the removeTreeFromNexus function throws an error when it should (_EDIT: testing d3 may be more trouble than it's worth in this instance; I'm going to add an issue to my backlog for getting the test for removeTreeFromNexus throwing an error to work because I think that would be good to know how to do in general and useful_)

There is also an open issue to account for partitions (https://github.com/veg/datamonkey-js/issues/81), but I don't think this should be a blocker for this PR.

Note that the first two commits in this PR are from #20 and will go away from the git diff when that is merged.

I also moved some of the components into a subdirectory to improve the file structure.

### CODE REVIEW

**Comment by rdvelazquez** [comment]
@stevenweaver This is now ready for your review.

File: src/components/job_submittal.jsx [hunk 1]
```diff
@@ -47,46 +49,66 @@ class JobSubmittal extends Component {
     this.setState({ [key]: value });
   };
 
+  saveBranchSelection = annotatedTree => {
+    this.updateJobInfo("annotatedTree", annotatedTree);
+    this.updateJobInfo("treePath", this.state.jobInfo.msaPath + ".tree");
+    ipcRenderer.send("saveAnnotatedTree", {
```
Comment (stevenweaver) [inline src/components/job_submittal.jsx]: This will need to be abstracted out to its own library

File: src/components/job_submittal.jsx [hunk 2]
```diff
@@ -111,16 +133,36 @@ class JobSubmittal extends Component {
...
+
+        {/* Branch Selection (if the method requires it) */}
+        {methodSpecificInfo[self.props.method].branchSelection &&
+        self.state.filePassedValidation &&
+        self.state.branchSelectionSaved == false ? (
+          <BranchSelection
+            userSuppliedNwkTree={this.state.jobInfo.tree.user_supplied}
+            neighborJoiningNwkTree={this.state.jobInfo.tree.neighbor_joining}
+            nwkTree={this.state.jobInfo.tree}
+            returnAnnotatedTreeCallback={this.saveBranchSelection}
```
Comment (stevenweaver) [inline src/components/job_submittal.jsx]: Very nice.

File: src/components/submittal_subcomponents/branch_selection.jsx [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        alert(
+          "No branch selections were made, please select one. Alternatively, you can choose to select all via the menu."
+        );
+      }
+    }
+
+    validate_selection(tree, callback);
+  }
+
+  render() {
```
Comment (stevenweaver) [inline src/components/submittal_subcomponents/branch_selection.jsx]: Super clean render function.

File: src/components/submittal_subcomponents/parse_and_validate_msa.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      }
+    });
+  };
+
+  render() {
+    return (
+      <div>
+        <button
+          onClick={() =>
+            ipcRenderer.send("parseAndValidateMSA", {
```
Comment (stevenweaver) [inline src/components/submittal_subcomponents/parse_and_validate_msa.jsx]: If we make all calls to ipcRenderer a call to a new library class, I have a feeling that we can utilize all of this code as a drop-in for datamonkey. I can see what I can do.

**Pull request merged**